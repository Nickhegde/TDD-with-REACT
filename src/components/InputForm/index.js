import React from 'react';
import { useForm } from 'react-hook-form';
import { STRINGS } from 'consts';

import './InputForm.scss';

export default function InputForm() {
  const { register, handleSubmit, watch, errors } = useForm();
  const onSubmit = data => console.log(data);

  return (
    <form className='form-container' onSubmit={handleSubmit(onSubmit)}>
      <input className='form-inputs' name="firstName" placeholder="First Name" ref={register({ required: true, maxlength: 20 })} />
      <span>{errors.firstName && 'First name is required'}</span>
      <input className='form-inputs' name="lastName" placeholder="Last Name" ref={register({ required: true, maxlength: 20 })} />
      <span>{errors.lastName && 'Last name is required'}</span>
      <select className='form-inputs' name="gender" ref={register}>
        <option value="male">Male</option>
        <option value="female">Female</option>
      </select>
      <input className='form-inputs' name="age" type="number" placeholder="Age" ref={register({ required: true, min: 10 })} />
      <input className='form-inputs' type="tel" placeholder="Mobile number" name="Mobile number" ref={register({ required: true, minLength: 0, maxLength: 10 })} />
      <button className='form-submit-btn' type="submit">{STRINGS.SUBMIT}</button>
    </form>
  );
}
