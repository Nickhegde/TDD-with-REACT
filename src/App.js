import React, { useState } from 'react';
import { InputForm } from './components';
import logo from './logo.svg';
import './App.css';

function App() {
  const [showForm, setShowForm] = useState(false);

  return (
    <div className="App">
      {showForm ? <InputForm></InputForm> :
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <div
            className="App-link"
            onClick={() => { setShowForm(true) }}
          >
            Show Form
        </div>
        </header>
      }
    </div>
  );
}

export default App;
