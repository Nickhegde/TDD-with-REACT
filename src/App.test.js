import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';
import { InputForm } from 'components';


describe('App component', () => {
  it('showForm equals to true when the App-link  button is clicked ', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('header').length === 1);
    const Btn = wrapper.find('div.App-link');
    Btn.simulate('click');
    expect(wrapper.find('header').length === 0);
    expect(wrapper).toMatchSnapshot();
  });
});