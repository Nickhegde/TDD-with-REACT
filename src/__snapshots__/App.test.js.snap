// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component showForm equals to true when the App-link  button is clicked  1`] = `ShallowWrapper {}`;
